src/libclipper/include/clipper/json_util.hpp

```diff
@@ -65,6 +74,20 @@ rapidjson::Value& check_kv_type_and_return(rapidjson::Value& d,
   return val;
 }
 
+/* Check for matching types else throw exception */
+rapidjson::Value& check_kv_type_is_bool_and_return(rapidjson::Value& d,
```

Comment:
Yep, that could also definitely work. Both of these solutions seem pretty hacky; I'll look into a cleaner fix.